app/models/service_template_transformation_plan_request.rb

```diff
@@ -15,6 +15,16 @@ def source_vms
     vm_resources.where(:status => [ServiceResource::STATUS_QUEUED, ServiceResource::STATUS_FAILED]).pluck(:resource_id)
   end
 
+  def validate_conversion_hosts
+    transformation_mapping.transformation_mapping_items.each do |item|
+      next unless %w(EmsCluster CloudTenant).include?(item.source_type)
+      ems = item.destination_type.constantize.find(item.destination_id).ext_management_system
+      conversion_hosts = ConversionHost.all.select { |ch| ch.ext_management_system == ems }
```

Comment:
Hm this is unfortunate, something like `ConversionHost.select { |ch| ch.ext_management_system == ems }` would be nicer but we should really add an ems_id to `ConversionHost` if we need to do this sort of thing so we don't need to load every conversion host.

Reply:
I think it'd be better to add associations to ext_management_system like:

```
  has_many :host_conversion_hosts, :through => :hosts, :source => :conversion_host
  has_many :vm_conversion_hosts, :through => :vms, :source => :conversion_host

  def conversion_hosts
    host_conversion_hosts + vm_conversion_hosts
  end
```